lib/rules/no-unused-prop-types.js

```diff
@@ -17,7 +17,7 @@ var annotations = require('../util/annotations');
 // Constants
 // ------------------------------------------------------------------------------
 
-var DIRECT_PROPS_REGEX = /^props\s*(\.|\[)/;
+var DIRECT_PROPS_REGEX = /^(props|nextProps)\s*(\.|\[)/;
```

Comment:
I pushed a try. I split the regex into 2 variables. It leads to more conditions. Tell me if you prefer this version.